core/jvm/src/test/scala/fs2/CompressSpec.scala

```diff
@@ -168,5 +513,454 @@ class CompressSpec extends Fs2Spec {
+      val longString
+          : String = Array.fill(1024 * 1024 + 1)("x").mkString("") // max(compress.fileNameBytesSoftLimit, compress.fileCommentBytesSoftLimit) + 1
+      var (expectedFileName, actualFileName) =
+        (Option(toEncodableFileName(longString)), Option.empty[String])
+      var (expectedComment, actualComment) =
+        (Option(toEncodableComment(longString)), Option.empty[String])
+      Stream
+        .chunk(Chunk.empty[Byte])
+        .through(compress.gzip[IO](8192, fileName = Some(longString), comment = Some(longString)))
+        .unchunk // ensure chunk sizes are less than file name and comment size soft limits
+        .through(compress.gunzip[IO](8192, fileNameCallback = (s: String) => {
+          actualFileName = Some(s)
+        }, commentCallback = (s: String) => { actualComment = Some(s) }))
+        .compile
+        .toVector
+        .asserting { _ =>
+          actualFileName
+            .map(_.length)
+            .getOrElse(0) should be < expectedFileName.map(_.length).getOrElse(0)
+          actualComment
+            .map(_.length)
+            .getOrElse(0) should be < expectedComment.map(_.length).getOrElse(0)
+        }
+    }
+
+    "unix.gzip |> gunzip" in {
+      var (expectedFilename, actualFilename) =
+        (Option(toEncodableFileName("fs2.compress")), Option.empty[String])
+      var (expectedComment, actualComment) = (Option.empty[String], Option.empty[String])
+      var (expectedModificationTime, actualModificationTime) =
+        (Some(Instant.parse("2020-02-04T22:00:02Z")), Option.empty[Instant])
+      val expectedContent = "fs2.compress implementing RFC 1952\n"
+      val compressed = Array(0x1f, 0x8b, 0x08, 0x08, 0x62, 0xe9, 0x39, 0x5e, 0x00, 0x03, 0x66, 0x73,
+        0x32, 0x2e, 0x63, 0x6f, 0x6d, 0x70, 0x72, 0x65, 0x73, 0x73, 0x00, 0x4b, 0x2b, 0x36, 0xd2,
+        0x4b, 0xce, 0xcf, 0x2d, 0x28, 0x4a, 0x2d, 0x2e, 0x56, 0xc8, 0xcc, 0x2d, 0xc8, 0x49, 0xcd,
+        0x4d, 0xcd, 0x2b, 0xc9, 0xcc, 0x4b, 0x57, 0x08, 0x72, 0x73, 0x56, 0x30, 0xb4, 0x34, 0x35,
+        0xe2, 0x02, 0x00, 0x57, 0xb3, 0x5e, 0x6d, 0x23, 0x00, 0x00, 0x00).map(_.toByte)
+      Stream
+        .chunk(Chunk.bytes(compressed))
+        .through(
+          compress.gunzip[IO](
+            modificationTimeCallback = (instant: Instant) => {
+              actualModificationTime = Some(instant)
+            },
+            fileNameCallback = (fileName: String) => { actualFilename = Some(fileName) },
+            commentCallback = (comment: String) => { actualComment = Some(comment) }
+          )
+        )
+        .compile
+        .toVector
+        .asserting { vector =>
+          actualFilename shouldBe expectedFilename
+          actualComment shouldBe expectedComment
+          actualModificationTime shouldBe expectedModificationTime
+          new String(vector.toArray, StandardCharsets.US_ASCII) shouldBe expectedContent
+        }
+    }
+
+    "gzip performance" - {
```

Comment:
Ditto above w.r.t. JMH and performance measurement.

Reply:
As above. Would you prefer I just remove these false performance tests along with the classic implementation?

Reply:
Yep! Obviously better to convert them into real JMH measurements, but I'm cool with just nuking them from orbit.

Reply:
Agreed. I'll replace these at the end with benchmarks...

Reply:
jmh:run -i 1 -f 1 -wi 1 fs2.benchmark.CompressBenchmark fs2.benchmark.Compress2Benchmark

bytes = 1 MiB
bufferSize = 32 kiB
withRandomBytes => random bytes or zero bytes

```
[info] Benchmark                   (withRandomBytes)   Mode  Cnt     Score   Error  Units

[info] Compress2Benchmark.deflate               true  thrpt         29.175          ops/s
[info] Compress2Benchmark.deflate              false  thrpt        169.255          ops/s
[info] Compress2Benchmark.inflate               true  thrpt       1317.750          ops/s
[info] Compress2Benchmark.inflate              false  thrpt        385.535          ops/s
[info] Compress2Benchmark.gzip                  true  thrpt         27.654          ops/s
[info] Compress2Benchmark.gzip                 false  thrpt        172.131          ops/s
[info] Compress2Benchmark.gunzip                true  thrpt       2431.867          ops/s
[info] Compress2Benchmark.gunzip               false  thrpt        384.544          ops/s

[info] CompressBenchmark.deflate                true  thrpt          6.849          ops/s
[info] CompressBenchmark.deflate               false  thrpt        160.612          ops/s
[info] CompressBenchmark.inflate                true  thrpt          7.231          ops/s
[info] CompressBenchmark.inflate               false  thrpt          7.352          ops/s
[info] CompressBenchmark.gzip                   true  thrpt         26.660          ops/s
[info] CompressBenchmark.gzip                  false  thrpt        171.284          ops/s
[info] CompressBenchmark.gunzip                 true  thrpt       1045.854          ops/s
[info] CompressBenchmark.gunzip                false  thrpt        461.132          ops/s
```